apps/src/templates/textResponses/textResponsesRedux.js

```diff
@@ -0,0 +1,125 @@
+import {PropTypes} from 'react';
+
+// Shape for an individual text response
+export const textResponsePropType = PropTypes.shape({
+  puzzle: PropTypes.number.isRequired,
+  question: PropTypes.string,
+  response: PropTypes.string.isRequired,
+  stage: PropTypes.string.isRequired,
+  studentId: PropTypes.number.isRequired,
+  studentName: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired
+});
+
+ // Initial state of textResponsesRedux
```

Comment:
could be useful to have a comment explaining that responseData is an object where the key is a script, and the value is etc.

In progress, we appended "byScript" to our names to make it a bit more clear: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/sectionProgress/sectionProgressRedux.js#L153